### DISCUSSION TITLE
Contribuindo para projetos Open Source no Github mesmo sendo iniciante

### DISCUSSION THREAD

**Discussion by woliveiras** [frontendbr/forum#1033] [Discussão]
Fala, galera!

A algum tempo escrevi um artigo sobre contribuição em projetos open source mesmo sendo iniciante.

Vejo que algumas pessoas continuam com essa dúvida, então gostaria de deixar aqui o link para o artigo e puxar a thread para conversarmos mais sobre isso.

Artigo: [Contribuindo para projetos Open Source no Github mesmo sendo iniciante](https://medium.com/trainingcenter/contribuindo-para-projetos-open-source-no-github-mesmo-sendo-iniciante-25a5a0ad6215)

**Como vocês, pessoas iniciantes, tem contribuído e como começaram a contribuir com projetos open source?**

**O que vocês acham que podemos fazer algo para facilitar a entrada de gente iniciante nos projetos?**

**Comment by willcezar00** [comment]
Tenho muita vontade de contribuir,  estou atualmente tentando me engajar e aprender mais,  porém trabalho tempo integral e tenho algumas outras obrigações no momento ( estou tira do a cnh e estudando reactjs),  estou tentando ajustar meu tempo,  mas gostaria de contribuir com algum projeto,  tentar me "enturmar",  e quem sabe compartilhar a absorver um pouco de conhecimento,  mudei um pouco minha vida ultimamente,  decide que iria mesmo que pouco,  me dedicar diariamente  a aprender algo novo,  comecei com as. Minhas redes sociais,  descurti e deixei de seguir tudo que não era relacionado a desenvolvimento,  então comecei um curso de reactjs,  e agora estou sendo mais ativos em conversar e discussões sobre assuntos relacionados  à tecnologia,  e atualmente procurando achar algo e em que possa ajudar a comunidade,  ainda não sei como fazer isso,  mas acho que se continuar no caminho que estou chegarei lá,  tenho experiência com desenvolvimento . NET (Mvc e webapi), angular 5,  um pouco em Java(Android e spring)  e reactjs,  se alguém souber de algo em que posso contribuir me dê um toque rsrs.  Obrigado

**Comment by emerson-d-lopes** [comment]
@willcezar00 to no meio que no mesmo caminho que voce! Uma dica que eu peguei uma vez, não me lembro onde, é de não procurar por nenhum projeto, tipo pega algo que você usa, um framework seja o que for ve o codigo do github, tenta melhorar algo que você acha que podia ser melhorado, sobre participar da comunidade simplesmente entra e vai dando pitaco haha. 

Eu ainda não achei nenhum projeto que eu tivesse conhecimento o necessario pra ajudar mas tambem tenho muita vontade creio que os primeiros são sempre mais complicados, pra entender como tudo funciona e tals.

**Comment by willcezar00** [comment]
Boa dica,  eu tô fazendo isso também,  tentando interagir e estudando o que posso,  estou ficando cada vez mais motivado,  e estou tentando otimizar meu tempo de forma geral,  reservando um tempo para tirar a cnh e para cuidar da saúde (academia),   isso tem me deixado Mais animado para fazer coisas e aprender. Mais.  Estou tentado trocar um estratégica e me ater a ela rsrs.

**Reply by Bezalleell** [reply]
sou iniciante, mas tenho muita dedicação nos estudos, eu evolui muito nesse ultimo ano, sei que participar de um projeto me trará mais experiencia

**Comment by felipefialho** [comment]
Alguns já devem ter visto [essa minha live na Alura](https://www.youtube.com/watch?v=0mi2pi4kkVk&t=2s) semana passada que tem exatamente esse tema! haha

É um dos meus assuntos favoritos e acho que de fato open-source pode contribuir pra acelerar a própria carreira e de quebra pode ajudar outras pessoas pela sua caracteristica aberta e colaborativa.

Considero open-source algo muito além de código e de fato penso que não é necessário codar para contribuir.

- Ajudar em eventos
- Ajudar grupos no slack
- Participar de fóruns
- Escrever artigos 

Essas coisas não exigem uma linha de código e possuem todos os fatores de colaboração exigidos em open-source. Mais do que isso, observe que participação em ONGs e projetos sociais TAMBÉM possuem essas caracteristicas colaborativas, ou seja tem gente participando de open-source e não faz ideia disso.

Especificamente no Github, também é possivel colaborar sem codar e mesmo sem saber git. Já que é possível submeter PRs de documentação alterando arquivos .md direto no site. 

Caso a ideia seja contribuir codando, acho que o melhor é procurar issues abertas em projetos que você já utiliza, já que provavelmente você vai ter conhecimentos básicos da linguagem utilizada e do proprio ecosistema.

Dúvidas estamos ai o/

**Comment by ninetails** [comment]
Bom, meu caso é bem parecido do @willcezar00 @lupuselit. Nunca fui atrás, talvez tenha motivação mas não vontade e diria que muito por ser tímido por ser julgado pela comunidade e outras coisas demotivacionais sobre minha pessoa.

Mas um dia desses subi uma pequena contribuição, foi bem recompensadora por uma primeira vez.

**Comment by ghost** [comment]
Acho que deveria ter um repositorio com links de projetos que aceitam commits, tipo o cara coloca no readme que precisa dá um tapa no seu site, ai tipo se tiver algum design que quer contribuir ele abre um pull request.

Eu tava esses dias procurando um projeto mas acabei mais me beneficiando do que ajudando. :P

**Comment by felipefialho** [comment]
@Rxon7 https://github.com/frontendbr/open-source

**Comment by ghost** [comment]
@LFeh top, só faltou umas label's com o nivel de conhecimento. #fikdik

**Comment by ninetails** [comment]
btw relacionado: https://github.com/MunGell/awesome-for-beginners

**Comment by gabrieldarezzo** [comment]
Relacionado ao mundo de OpenSource

@zenorocha fez uma palestra bem legal:
"Por que ninguém se importa com seu novo projeto open source?" 
https://www.youtube.com/watch?v=l2rVPHCO5BA

**Comment by weltonsanches** [comment]
@Rxon7 além da lista que o @LFeh citou tem essa do TC https://github.com/training-center/divertidalista

> só faltou umas label's com o nivel de conhecimento

Essa ta organizada por nivel de conhecimento https://github.com/camilatigre/listamaravilhosaopensource

**Comment by danielschmitz** [comment]
Quem quiser contribuir traduzindo um dos documentos do VueJS Cookbook, aí está:

https://github.com/vuejs-br/br.vuejs.org/labels/help%20wanted

Lembrando que a sua contribuição vai pro site oficial do vuejs.org (translations>portuuese)

**Comment by munizart** [comment]
Um bom lugar pra achar issues pra começar é a pesquisa por label do github, um exemplo:
[`is:open label:good first issue`](https://github.com/search?l=JavaScript&q=is%3Aopen+label%3A%22good+first+issue%22+&type=Issues).
Acho que da pra fazer um site estatico consumindo essa e outras busca e mostrando um agregado das labels.
Alguem topa? (ou principalmente: alguem sabe se já existe?) :D

**Comment by danielschmitz** [comment]
@munizart  https://github.com/explore pode ser um bom começo

**Comment by Marlysson** [comment]
@munizart tem esse por exemplo:

https://24pullrequests.com/

Nesse post por exemplo ele mostra 3 sites diferentes contendo listas de projetos open source que esperam contribuição:

https://blog.kentcdodds.com/what-open-source-project-should-i-contribute-to-7d50ecfe1cb4

- Up for Grabs
- 24 Pull Requests

**Comment by synga** [comment]
Gosto de pegar uns projetos pequenos pra ajudar, galera que tem umas ideias legais e precisam de uma ajuda com alguma coisa. Pra projetos maiores não fiz muita coisa relacionada a código, mas sempre abro umas issues tentando explicar ao máximo algum comportamento, como reproduzir um erro e etc., também fico em uns canais de slack de projetos incentivando discussões, acho bacana a galera parar pra refletir, pensar nas possibilidades e tentar deixar passar batido o menor numero de coisas possíveis, por enquanto essas são as formas que me sinto confortável pra contribuir.

Mas acho que um modo bacana de ajudar quem está começando seria colocar nos repositórios o que a galera pode ajudar, um _disclaimer_ ou _help wanted_ falando o que é possível ajudar naquele projeto como tradução, correção de texto, issues, código, slack, design e tudo mais que for possível. Tem muita gente que quer começar a contribuir, mas acha que se resume apenas a código.

Outra coisa seria ter um arquivo legal de _contributing_ e colocar no _readme_ que existe esse arquivo, que existem essas regras, não só pra galera saber como contribuir de forma certa, mas também para aprenderem a seguir alguns padrões e tal. Talvez de até pra fazer essa primeira ideia no arquivo de _contributing_.

**Comment by alessandroholiver** [comment]
Sou novato na área. Gostaria de ajudar em algum código simples e com isso aprender. Sou estudante do 4º periodo de Sistemas de informação.
Como devo começar?

**Comment by synga** [comment]
@alessandroholiver Cara, vou te falar que aprendi mais contribuindo de outras formas do que codando, então vou te passar o que funcionou pra mim e se alguém quiser e se sentir a vontade poderá te informar sobre repositórios pra aprender codando.

Uma coisa que eu fiz foi focar em alguma tecnologia/framework e criar um curso. Ler bastante, estudar, entender, ver como funciona, o que acontece por trás, diferenças para outras tecnologias e tal. Gravei umas vídeo aulas e lancei. Fora que isso melhorou MUITO minha questão de comunicação, conseguir me expressar, passar minhas ideia de forma um pouco mais claras e trabalhar varias outras das chamadas _soft skills_.

Outra solução foi traduzir conteúdo, corrigir traduções e tudo mais. Se você conseguir ler e entender inglês já era, não precisa ser fluente ou qualquer coisa do tipo. Pega repositórios de iniciativas de ensino, projetos que tem traduções e coisas assim (como o [FreeCodeCamp](https://github.com/freeCodeCamp/freeCodeCamp)) e traduza umas coisas. Além de exercitar um outro idioma você vai estar aprendendo provavelmente com gente foda, que escreve bem, ensina bem e tem uns macetes legais. Já mata duas coisas de uma vez.

Enfim, isso ai me rendeu um conhecimento legal e vários aprendizados diferentes. Mas se quiser realmente tentar apenas codando da um tempinho que logo logo alguém te passa umas dicas.

**Comment by alessandroholiver** [comment]
Muito obrigado. 
Irei ler mais, correr atrás. O unico contato que estou tendo com a programação é na faculdade.
para consegui o primeiro emprego de desenvolvedor par um cara de 33 anos é um pouco dificil.

Mas, nao irei desistir.

**Comment by felipefialho** [comment]
@alessandroholiver temos o https://github.com/frontendbr/open-source com alguns projetos brasileiros precisando de ajuda. Dá uma olhadinha lá.

**Comment by mutheusalmeida** [comment]
@alessandroholiver, pesquise sobre a freeCodeCamp. Ela me ajudou bastante e com certeza irá ajudar você também.

**Comment by ghost** [comment]
Ótimas informações

**Comment by laurianops** [comment]
@woliveiras quero saber como faço para contribuir traduzindo o codigo, eu tenho que traduzir o README.md ? fazer outro ? ou como é ? 
abraços

**Comment by ghost** [comment]
> @woliveiras quero saber como pra contribuir traduzindo o codigo, eu tenho que traduzir o README.md ? fazer outro ? ou como é ?
> abraços

Cara você pode fazer um pull request ou forkar ou pode fazer um commit.

**Comment by woliveiras** [comment]
@LaurianoElmiroDuarte tradução é uma excelente ajuda. Além de ajudar o projeto em si, você atinge milhares de pessoas que não tem acesso ao idioma em que a ferramenta está desenvolvida. 

Você pode traduzir documentações, artigos, palestras ou adicionar legendas em vídeos.

Normalmente os projetos trem uma documentação no próprio GitHub, daí podemos fazer um fork, como o @pedroguilhermelima comentou e só ir traduzindo. Mas nos outros casos você pode traduzir os artigos, vídeos, palestras, etc e colocar no seu blog pessoal ou no blog de alguma comunidade. :) 

O iMasters sempre aceita novos escritores, inclusive.

**Comment by laurianops** [comment]
entendi então vai direto ne @pedroguilhermelima  ..
eu farei sim @woliveiras 
obrigadão a todos.

**Comment by jbbn** [comment]
vou tentar deixar meus _2 cents_ ;)

## Colaborar com coisas "mais simples", ex. tradução

O mais interessante de fazer **coisas simples** é que por elas serem "simples" você está praticando outras coisas além da produção de código. Afinal, no cotidiano, a gente tem várias atribuições além de produzir código.

Quando você contribui para um projeto você vai praticar coisas que, talvez, esteja fazendo pela primeira vez, como:

* Tirar dúvidas em uma issue
* Realizar um fork
* Fazer uma PR (Pull Request)
* Ajustar uma PR que foi revisada e **mudanças foram solicitadas**
* Interagir com os comentários que foram feitos na PR
* (Talvez ajustar a história do seu branch para clareza de histórico)
* (Ver seu código no publicado mostrar e contar para seus colegas de trabalho [talvez para sua família {e perceber o quão difícil é explicar para eles a importãncia e/ou _legalzeza_ do que você fez}])

Aí você vai contribuir pela segunda vez, com uma coisa simples (para o mesmo repositório), só que entraram outras coisas nesse meio tempo e adivinha? Seu fork está desatualizado rs. É uma boa hora para aprender a atualizar seu fork com as atualizações do repositório pai ;)

Começar com coisas simples é ótimo ;) Traduza mais de uma vez para o mesmo repo, você só tem a ganhar com isso.

## Conselho: leia o código

Leia o código do repositório. Releia. Tenha certeza que você entendeu como o código funciona. Você vai aprender:

* Qual o entrypoint do projeto? (cada projeto tem um jeito diferente de executar seu propósito)
* Como rodar o projeto para desenvolvimento?
* Como realizar a build do projeto?
* Como a build **realmente** funciona? (você vai se deparar com maneiras diferentes de fazer as coisas; será bom para sua experiẽncia, pois com abordagens diferentes para fazer as mesmas coisas, você começa a perceber prós e contras de uma e de outra; sem perceber você vai olhar para um projeto e pensar "acho que no outro projeto que vi tinha uma abordagem melhor para esse propósito" [com isto, talvez seja o caso de abrir uma issue e perguntar aos mantenedores o que eles acham {aí eles te respondem "legal, você poderia fazer uma PR para vermos melhor como ficaria?" aí você sente que faz parte de algo maior e tem um ataque cardíco.. brincadeira.. é só euforia mesmo}])
* Como organizar o código? (módulos, libs, helpers, content, scripts, docs, etc...)

Quando eu digo "Leia o código" eu quero dizer para você revisar e aprender com ele. Não passe só o olho e diga "ah tah". Tenha certeza que você entendeu o que cada linha faz. Novamente, você aprende maneiras diferentes de fazer a mesma coisa. E isso é o ínicio de muitas contribuições.

O que acontece, às vezes, com alguns repositórios modulares que já tem algum tempo, é que algum módulo "carrega" mais experiência no código (aquela abordagem melhorada que citei antes). Uma boa contribuição, pode ser melhorar um módulo do repo que é mais antigo, mas se beneficiaria das melhores práticas descobertas depois. Além disso, é uma maneira excelente de você praticar o que aprendeu com um cenário mais próximo do real.

----

Eu provavelmente me excedi um pouco rs - mas as dicas iniciais são essas. Lendo código aberto e contribuindo um pouco aqui e ali **transformou** meus conhecimentos e experiẽncia. Quem sabe não faz o mesmo por mais alguém.

Abraços

**Comment by ghost** [comment]
> Boa dica, eu tô fazendo isso também, tentando interagir e estudando o que posso, estou ficando cada vez mais motivado, e estou tentando otimizar meu tempo de forma geral, reservando um tempo para tirar a cnh e para cuidar da saúde (academia), isso tem me deixado Mais animado para fazer coisas e aprender. Mais. Estou tentado trocar um estratégica e me ater a ela rsrs.

Eu estou tentando passar um repositório para o inglês, você pode me ajudar ? ;D

**Comment by ghost** [comment]
> vou tentar deixar meus _2 cents_ ;)
> 
> ## Colaborar com coisas "mais simples", ex. tradução
> O mais interessante de fazer **coisas simples** é que por elas serem "simples" você está praticando outras coisas além da produção de código. Afinal, no cotidiano, a gente tem várias atribuições além de produzir código.
> 
> Quando você contribui para um projeto você vai praticar coisas que, talvez, esteja fazendo pela primeira vez, como:
> 
> * Tirar dúvidas em uma issue
> * Realizar um fork
> * Fazer uma PR (Pull Request)
> * Ajustar uma PR que foi revisada e **mudanças foram solicitadas**
> * Interagir com os comentários que foram feitos na PR
> * (Talvez ajustar a história do seu branch para clareza de histórico)
> * (Ver seu código no publicado mostrar e contar para seus colegas de trabalho [talvez para sua família {e perceber o quão difícil é explicar para eles a importãncia e/ou _legalzeza_ do que você fez}])
> 
> Aí você vai contribuir pela segunda vez, com uma coisa simples (para o mesmo repositório), só que entraram outras coisas nesse meio tempo e adivinha? Seu fork está desatualizado rs. É uma boa hora para aprender a atualizar seu fork com as atualizações do repositório pai ;)
> 
> Começar com coisas simples é ótimo ;) Traduza mais de uma vez para o mesmo repo, você só tem a ganhar com isso.
> 
> ## Conselho: leia o código
> Leia o código do repositório. Releia. Tenha certeza que você entendeu como o código funciona. Você vai aprender:
> 
> * Qual o entrypoint do projeto? (cada projeto tem um jeito diferente de executar seu propósito)
> * Como rodar o projeto para desenvolvimento?
> * Como realizar a build do projeto?
> * Como a build **realmente** funciona? (você vai se deparar com maneiras diferentes de fazer as coisas; será bom para sua experiẽncia, pois com abordagens diferentes para fazer as mesmas coisas, você começa a perceber prós e contras de uma e de outra; sem perceber você vai olhar para um projeto e pensar "acho que no outro projeto que vi tinha uma abordagem melhor para esse propósito" [com isto, talvez seja o caso de abrir uma issue e perguntar aos mantenedores o que eles acham {aí eles te respondem "legal, você poderia fazer uma PR para vermos melhor como ficaria?" aí você sente que faz parte de algo maior e tem um ataque cardíco.. brincadeira.. é só euforia mesmo}])
> * Como organizar o código? (módulos, libs, helpers, content, scripts, docs, etc...)
> 
> Quando eu digo "Leia o código" eu quero dizer para você revisar e aprender com ele. Não passe só o olho e diga "ah tah". Tenha certeza que você entendeu o que cada linha faz. Novamente, você aprende maneiras diferentes de fazer a mesma coisa. E isso é o ínicio de muitas contribuições.
> 
> O que acontece, às vezes, com alguns repositórios modulares que já tem algum tempo, é que algum módulo "carrega" mais experiência no código (aquela abordagem melhorada que citei antes). Uma boa contribuição, pode ser melhorar um módulo do repo que é mais antigo, mas se beneficiaria das melhores práticas descobertas depois. Além disso, é uma maneira excelente de você praticar o que aprendeu com um cenário mais próximo do real.
> 
> Eu provavelmente me excedi um pouco rs - mas as dicas iniciais são essas. Lendo código aberto e contribuindo um pouco aqui e ali **transformou** meus conhecimentos e experiẽncia. Quem sabe não faz o mesmo por mais alguém.
> 
> Abraços

Massa, que dica! ^^

**Comment by jbbn** [comment]
obrigado @pedroguilhermelima 

Feedback sempre é motivador. Estou tentando participar mais desta comunidade de frontends ;)

**Comment by ghost** [comment]
> obrigado @pedroguilhermelima
> 
> Feedback sempre é motivador. Estou tentando participar mais desta comunidade de frontends ;)

Valeu ;D Vou ver se ajudo você também ^^

**Comment by dmchiodini** [comment]
Sensacional! Me abriu a mente, como sou iniciante ficava meio perdido em relação a isso. Obrigado!

**Comment by ghost** [comment]
Eu fiz o projeto opensource "ReviewReact".. seria minha ideia de um novo formato de mídia... colaborem se quiserem ^^ teria a finalidade de ajudar no desenvolvimento ao mesmo tempo rindo e percebendo os erros comuns para melhorar o design etc...

**Comment by Brutos212** [comment]
Bom sou bem novato no assunto, estou estudando Pyhton, quase dia e noite, será interessante encontrar algum projeto

**Comment by ghost** [comment]
> Bom sou bem novato no assunto, estou estudando Pyhton, quase dia e noite, será interessante encontrar algum projeto

Sim. Rs... começa **^^**

**Comment by vinic1usdev** [comment]
A melhor forma de um iniciante entrar no mercado de trabalho e conseguir é trabalha em projetos open sources , essa é minha opinião, mesmo que não contribua, mas analisar os arquivo lê os comentários isso gera uma grande conhecimento,  e isso é que eu estou buscando.
